packages/syft/src/syft/service/sync/resolve_widget.py

```diff
@@ -110,17 +110,24 @@ def __init__(
         self.with_box = with_box
         self.widget = self.build()
         self.sync = True
-        self.mockify = False
-        if isinstance(diff.non_empty_object, TwinAPIEndpoint):
-            self.share_private_data = False
-        else:
-            self.share_private_data = True  # there is no private data in this case
         self.is_main_widget: bool = True
 
     def set_share_private_data(self) -> None:
         # No-op for main widget
         pass
 
+    @property
+    def mockify(self) -> bool:
+        # mock TwinAPIEndpoint.__private_sync_attr_mocks__
+        if isinstance(self.diff.non_empty_object, TwinAPIEndpoint):
+            return True
+        return isinstance(self.diff.non_empty_object, TwinAPIEndpoint)
+
+    @property
+    def share_private_data(self) -> bool:
```

Comment:
shouldnt the logic be the other way around? If we dont share private data, we dont mockify?